operator/api/redpanda/v1alpha2/common.go

```diff
@@ -271,3 +272,32 @@ func (c *ClusterSource) GetAdminAPISpec() *AdminAPISpec {
 func (c *ClusterSource) GetClusterRef() *ClusterRef {
 	return c.ClusterRef
 }
+
+const (
+	ResourceConditionTypeSynced = "Synced"
+
+	ResourceConditionReasonPending              = "Pending"
+	ResourceConditionReasonSynced               = "Synced"
+	ResourceConditionReasonClusterRefInvalid    = "ClusterRefInvalid"
+	ResourceConditionReasonConfigurationInvalid = "ConfigurationInvalid"
+	ResourceConditionReasonTerminalClientError  = "TerminalClientError"
```

Comment:
Maybe, I use the notion of a "terminal" error in two places currently, which is how the `UserReconciler` works:

1. https://github.com/redpanda-data/redpanda-operator/blob/f61b187e0bcd7e5e4ffad11f0d10a3b8b4438d2e/operator/internal/controller/redpanda/resource_controller.go#L122-L137
2. https://github.com/redpanda-data/redpanda-operator/blob/f61b187e0bcd7e5e4ffad11f0d10a3b8b4438d2e/operator/internal/controller/redpanda/resource_controller.go#L139-L153

The first is basically if you want to ignore errors that shouldn't ever be retried (i.e. you have a non-existent cluster that's been blown away and you're trying to delete yourself from it, cluster configuration now has SASL auth disabled and you're trying to delete a user, etc.). This is used during the cleanup routine to do "best-effort" style cleanup, if something is "terminal" we basically just say, "oh well" and let the resource be GC'd if we hit one of these.

The second is in the case of you trying to sync a resource. In that case we just set the status as having encountered a terminal error with the error message in the status condition message and then stop reconciling because this isn't an "ephemeral"-type error.

Reply:
Also 👍 on documenting this stuff more.